### PR TITLE
New /v3 cluster methods design, more mock functions

### PR DESCRIPTION
**Pull request by marians**
This change contains the design doc for two new API methods:

- `GET /v3/orgs/:orgname/clusters/`: Returns a list of all cluster for the current user
- `GET /v3/clusters/id/`: Returns details for one cluster
- `POST /v3/clusters/:id/owners/`: Add owner organizations to a cluster

### CODE REVIEW

**Comment by marians** [comment]
RFR @zeisss

File: README.md [hunk 1]
```diff
@@ -1,13 +1,71 @@
...
+
+List all clusters the current user has access to.
+
+Example output:
+
+```json
+{
+  "status_code": 10000,
+  "status_text": "success",
+  "data": [
```
Comment (zeisss) [inline README.md]: IMO we should get rid of this, but we should probably discuss this with the whole team.

File: README.md [hunk 2]
```diff
@@ -1,13 +1,71 @@
...
+
+```json
+{
+  "status_code": 10000,
+  "status_text": "success",
+  "data": {
+    "id": "x882ofna6ferso",
+    "create_date": "2016-07-01T12:00:00.000Z",
+    "name": "default",
+    "server": "https://x882ofna6ferso.giantswarm-kaas.io",
```
Comment (zeisss) [inline README.md]: I think we should call this field `apiserver`

File: README.md (hunk 1 above)
Comment (marians) [inline README.md]: Removed it.

File: README.md (hunk 2 above)
Comment (marians) [inline README.md]: Calling it `api_server` now

**Comment by marians** [comment]
Updated the draft after @zeisss gave first feedback.

File: README.md [hunk 3]
```diff
@@ -1,13 +1,81 @@
...
+```json
+{
+  "id": "x882ofna6ferso",
+  "create_date": "2016-07-01T12:00:00.000Z",
+  "name": "default",
+  "api_server": "https://x882ofna6ferso.giantswarm-kaas.io",
+  "service_account": [
+    {
+      "name": "giantswarm-default-admin",
+      "client_certificate_data": "<CLIENT_CERTIFICATE_DATA>",
```
Comment (hectorj2f) [inline README.md]: I am not sure of this data in terms of security...

File: README.md [hunk 4]
```diff
@@ -1,13 +1,81 @@
...
+#### `GET /v3/clusters/<id>/`
+
+Example output for `GET /v3/clusters/x882ofna6ferso/`:
+
+```json
+{
+  "id": "x882ofna6ferso",
+  "create_date": "2016-07-01T12:00:00.000Z",
+  "name": "default",
+  "api_server": "https://x882ofna6ferso.giantswarm-kaas.io",
```
Comment (hectorj2f) [inline README.md]: I will called `api_endpoint`

Comment (marians) [inline README.md]: Will do

File: README.md (hunk 3 above)
Comment (marians) [inline README.md]: That will have to be discussed elsewhere.

**Comment by marians** [comment]
@zeisss @teemow @hectorj2f I've just updated the draft in `README.md`, adding the function `POST /v3/clusters/:id/owners/` for adding owners.

Please note that I suggest here a change in singular/plural naming, to fix something which has always made me mad in our API.

- `/v3/orgs/`: Here you find everything regarding all organizations, hence plural
- `/v3/orgs/:orgname/clusters/`: Here you find all the clusters associatd with the organization
- `/v3/clusters/`: Here you deal with all the clusters, hence plural
- `/v3/clusters/:id/owners/`: Here you deal with all the cluster owners, hence plural

Bottom line: Let's use plural from `v3` on (instead of singular like `/v3/org/:orgname/`etc.).

File: README.md [hunk 5]
```diff
@@ -1,13 +1,99 @@
...
+#### `POST /v3/clusters/:id/owners/`
+
+This method allows to add one or more organizations to the owners of a cluster.
+
+Example request body for `POST /v3/clusters/x882ofna6ferso/owners/`:
+
+```json
+{
+  "owners": [
+    "owner_type": "organization",
```
Comment (zeisss) [inline README.md]: do we still need this, as we just said we only want orgs here anyway?

**Comment by zeisss** [comment]
I am fine with this, although I think we should add a ticket to extend this with labels/annotations in the future, so we don't forget this.

File: README.md (hunk 5 above)
Comment (marians) [inline README.md]: I have a feeling that it would be cleaner and more extensible if we would explicitly mention that we are dealing with an org here.
